app/actions/service_instance_share.rb

```diff
@@ -3,18 +3,9 @@
 module VCAP::CloudController
   class ServiceInstanceShare
     def create(service_instance, target_spaces, user_audit_info)
-      supported_service_type?(service_instance)
-      service_instance_shareable?(service_instance)
-
-      if target_spaces.include?(service_instance.space)
-        raise CloudController::Errors::ApiError.new_from_details('InvalidServiceInstanceSharingTargetSpace')
-      end
-
-      target_spaces.each do |space|
-        if space.service_instances.map(&:name).include?(service_instance.name)
-          raise CloudController::Errors::ApiError.new_from_details('SharedServiceInstanceNameTaken', service_instance.name, space.name)
-        end
-      end
+      supported_service_type!(service_instance)
```

Comment:
Sounds good, we will rename it to `validate_supported_service_type!` since we throw different errors based on what type of service it is, and it's feels a bit cleaner to leave this check inside the method.

Reply:
Changes for this included in #1033 and can be viewed here: https://github.com/cloudfoundry-incubator/cloud_controller_ng_sapi/commit/ff8276e5d201e573df03640a0f6865288b2d55af#diff-e3c96f58c4686b95721452ad906b78a9R6